src/main/java/com/google/sps/servlets/Caption.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.InputStream;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class Caption {
+  /*
+   * Usage: String caption = new Caption().getCaptionFromID("abcdefg");
+   * 
+   * Return null if not found.
+   */
+
+  public CaptionService captionService;
+
+  public Caption() {
+    captionService = new CaptionService();
+  }
+
+  public void setCaptionService(CaptionService captionService) {
+    this.captionService = captionService;
+  }
+
+  public String getCaptionFromId(String videoId) {
+    // BCP47 language tags
+    final String[] languages = {"en", "en-US", "en-GB", "en-AU", "en-IE", "en-ZA"};
+    for (String language : languages) {
+      try {
+        return parseXmlFromStream(captionService.fetchStream(videoId, language));
+      } catch (SAXException | ParserConfigurationException | IOException
+          | IllegalArgumentException e) {
+        continue;
+      }
+    }
+    return null;
+  }
+
+  private String parseXmlFromStream(InputStream inputStream)
+      throws SAXException, ParserConfigurationException, IOException, IllegalArgumentException {
```

Comment:
Question: Is explicitly saying that these methods could throw these exceptions needed if this method is not the one creating the exception (only underlying methods are doing so). I would think you didn't need to do it because then every method in a chain of calls would need them and all methods in existence would need them lol.

Reply:
Thats a good counterexample. Can I generalize them as one Exception, since they will be ignored anyway in `getCaptionFromId()`?

Reply:
I guess you do need them. I guess keep as it is now. There are two types of exceptions as explained below.
https://stackoverflow.com/questions/11589302/why-is-throws-exception-necessary-when-calling-a-function

Now I know why you don't need to say that you throw exceptions like `IllegalArgumentException`

I don't recommend you change it to Exception because it makes it vague to callers. Also, this method doesn't know the caller wants to ignore them.

Reply:
okie I will keep it as it is. Thanks!